custom_components/tahoma/sensor.py

```diff
@@ -1,116 +1,338 @@
+        device_class=sensor.DEVICE_CLASS_SIGNAL_STRENGTH,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="core:ExpectedNumberOfShowerState",
+        name="Expected Number Of Shower",
+        icon="mdi:shower-head",
+        value=lambda value: round(value),
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="core:NumberOfShowerRemainingState",
+        name="Number of Shower Remaining",
+        icon="mdi:shower-head",
+        value=lambda value: round(value),
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    # V40 is measured in litres (L) and shows the amount of warm (mixed) water with a temperature of 40 C, which can be drained from a switched off electric water heater.
+    OverkizSensorDescription(
+        key="core:V40WaterVolumeEstimationState",
+        name="Water Volume Estimation at 40 °C",
+        icon="mdi:water",
+        value=lambda value: round(value),
+        unit_of_measurement=VOLUME_LITERS,
+        entity_registry_enabled_default=False,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="core:WaterConsumptionState",
+        name="Water Consumption",
+        icon="mdi:water",
+        value=lambda value: round(value),
+        unit_of_measurement=VOLUME_LITERS,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="io:OutletEngineState",
+        name="Outlet Engine",
+        icon="mdi:fan-chevron-down",
+        value=lambda value: round(value),
+        unit_of_measurement=VOLUME_LITERS,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="io:InletEngineState",
+        name="Inlet Engine",
+        icon="mdi:fan-chevron-up",
+        value=lambda value: round(value),
+        unit_of_measurement=VOLUME_FLOW_RATE_CUBIC_METERS_PER_HOUR,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="hlrrwifi:RoomTemperatureState",
+        name="Room Temperature",
+        value=lambda value: round(value),
+        device_class=sensor.DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="io:MiddleWaterTemperatureState",
+        name="Middle Water Temperature",
+        value=lambda value: round(value),
+        device_class=sensor.DEVICE_CLASS_TEMPERATURE,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="io:PriorityLockOriginatorState",
+        name="Priority Lock Originator",
+        icon="mdi:alert",
+    ),
+    OverkizSensorDescription(
+        key="core:FossilEnergyConsumptionState",
+        name="Fossil Energy Consumption",
+        device_class=sensor.DEVICE_CLASS_ENERGY,
+    ),
+    OverkizSensorDescription(
+        key="core:GasConsumptionState",
+        name="Gas Consumption",
```

Comment:
No device class?

Reply:
afaik, there is no device class for gas consumption sensors.